collect_app/src/main/java/org/odk/collect/android/utilities/FileUtil.java

```diff
@@ -19,4 +24,36 @@ public String getRandomFilename() {
     public File getItemsetFile(String mediaFolderPath) {
         return new File(mediaFolderPath + "/itemsets.csv");
     }
+
+    public static String getFileContents(final File smsFile) throws IOException {
+
+        final InputStream inputStream = new FileInputStream(smsFile);
+        final BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+
+        final StringBuilder stringBuilder = new StringBuilder();
+
+        boolean done = false;
+
+        while (!done) {
+            final String line = reader.readLine();
+            done = (line == null);
+
+            if (line != null) {
+                stringBuilder.append(line);
+            }
+        }
+
+        reader.close();
+        inputStream.close();
+
+        return stringBuilder.toString();
+    }
+
+    public static String getSmsInstancePath(String instancePath) {
+        StringBuilder builder = new StringBuilder();
+        builder.append(instancePath)
+                .append(".txt");
+
+        return builder.toString();
```

Comment:
It may just return `instancePath + ".txt"` right? 
I mean the whole method might be just:

>     public static String getSmsInstancePath(String instancePath) {
>         return instancePath + ".txt";
>     }

Reply:
Yes. There's no performance gain to be had here.